File: Homework/Mary_Antonyan/C++/30_03_2017/Chess/king.cpp
#include "king.hpp"
#include "board.hpp"
#include <iostream>

//Constructor
Chess::King::King(const Chess::Figure::Chessman figure, 
                    const Chess::Figure::Color color, 
                    const int row, const int coloumn) 
    : Figure(figure, color, row, coloumn)
{}

//Destructor
Chess::King::~King() {}

//Method override
void Chess::King::step(const int row, const int coloumn) {
    if (0 <= row && 8 > row && 0 <= coloumn && 8 > coloumn) { 
        if ((row == mRow - 1 || row == mRow + 1 || row == mRow) && 
            (coloumn == mRow - 1 || coloumn == mRow + 1 || coloumn == mColoumn) &&
            (Chess::board[row][coloumn] == nullptr || 
             Chess::board[row][coloumn]->getColor() != mColor)) {
            std::cout << "Move to [" << row << " , " << coloumn << "]\n";
            int tmpRow = mRow;
            int tmpCol = mColoumn;
            Chess::board[row][coloumn] = new King(mFigure, mColor, row, coloumn);
            Chess::board[tmpRow][tmpCol] = nullptr;
        } else {
            std::cout << "Can't move to [" << row << " , " << coloumn << "]\n";
        }
    } else {
        std::cout << "Out of board" << std::endl;
    }
}

File: Homework/Mary_Antonyan/Java/07_08_2017/Chess/src/main/java/com/itc/chess/King.java
package com.itc.chess;
import java.awt.Point;

/**
 * This class represents the king(թագավոր) chess figure
 *
 * @author mary
 */
public class King extends Figure {
    /**
     * Creates a king object
     * 
     * @param figure figure name
     * @param color figure color
     * @param position figure current position
     */
    public King(Chessman figure, Color color, Point position) {
        super(figure, color, position);
    }
    
    /**
     * 
     * @param position new position
     * @return can the object step to new position or not
     */
    @Override
    public boolean step(Point position) {
        if (0 <= position.x && 8 > position.x && 0 <= position.y && 8 > position.y) { 
            if (((position.x == this.position.x - 1 || position.x == this.position.x + 1 || position.x == this.position.x) && 
            (position.y == this.position.x - 1 || position.y == this.position.x + 1 || position.y == this.position.y) && 
            Board.board[position.x][position.y] == null || Board.board[position.x][position.y].getColor() != color)) {
                System.out.println("Move to [" + position.x + " , " + position.y + "]");
                int tmpX = this.position.x;
                int tmpY = this.position.y;
                Board.board[position.x][position.y] = new King(figure, color, position);
                Board.board[tmpX][tmpY] = null;
                return true;
            } else {
                System.out.println("Can't move to [" + position.x + " , " + position.y + "]");
            }
        } else {
            System.out.println("Out of board");
        }
        return false;
    }
}